content/jsoc/gsoc/sciml.md

```diff
@@ -0,0 +1,176 @@
+
+### Multi-GPU training
+
+Implement and demonstrate multi-GPU parallelism. One route is to expose communication primitives from NVIDIA's [NCCL](https://developer.nvidia.com/nccl) library and use these to build tooling for model parallelism and distributed training. The project should demonstrate parallel training of a Flux model with benchmarks.
+
+Mentors: [Valentin Churavy](https://github.com/vchuravy), [Tim Besard](https://github.com/maleadt)
+
+### Distributed Training
+
+Add a distributed training API to Flux, possibly inspired by PyTorch's equivalent. Any distributed training algorithm could be used (ideally the foundations make it easy to experiment with different setups), though the easiest is likely to implement an MXNet-like parameter server. It should demonstrate training a Flux model with data distributed over multiple nodes, with benchmarks.
+
+Mentors: [Valentin Churavy](https://github.com/vchuravy), [Tim Besard](https://github.com/maleadt)
+
+### Sparse GPU and ML support
+
+While Julia supports dense GPU arrays well via [CuArrays](https://github.com/JuliaGPU/CUSPARSE.jl), we lack up-to-date wrappers for sparse operations. This project would involve wrapping CUDA's sparse support, with [CUSPARSE.jl](https://github.com/JuliaGPU/CUSPARSE.jl) as a starting point, adding them to CuArrays.jl, and perhaps demonstrating their use via a sparse machine learning model.
+
+### NLP Tools and Models
+
+Build deep learning models for Natural Language Processing in Julia. [TextAnalysis](https://github.com/juliatext/TextAnalysis.jl)  and [WordTokenizers](https://github.com/JuliaText/WordTokenizers.jl) contains the basic algorithms and data structures to work with textual data in Julia. On top of that base, we want to build modern deep learning models based on recent research. The following tasks can span multiple students and projects.
+
+It is important to note that we want practical, usable solutions to be created, not just research models. This implies that a large part of the effort will need to be in finding and using training data, and testing the models over a wide variety of domains. Pre-trained models must be available to users, who should be able to start using these without supplying their own training data.
+
+* Implement [ELMo](https://allennlp.org/elmo) in Julia
+* Implement [ALBERT](https://ai.googleblog.com/2019/12/albert-lite-bert-for-self-supervised.html) in Julia
+* Implement GPT/GPT-2 in Julia
+* Implement [extractive summarisation based on Transformers](https://arxiv.org/abs/1909.03186)
+* Implement practical models for 
+  * Dependency Tree Parsing  
+  * Morphological extractions  
+  * Translations (using Transformers)  
+* Indic language support -- validate and test all models for Indic languages
+  * ULMFiT models for Indic languages
+* Chinese tokenisation and parsing
+
+**Mentors**: [Avik Sengupta](https://github.com/aviks/)
+
+### Automated music generation
```

Comment:
ML

Reply:
Are you pointing to ### Automated music generation?

Reply:
or the post above it?

Reply:
This one clearly goes to ML.